### ISSUE TITLE
Emacsclient fails with errormessage

### ISSUE THREAD

**Issue by mok0** [wez/wezterm#5770]
### What Operating System(s) are you seeing this problem on?

Linux X11

### Which Wayland compositor or X11 Window manager(s) are you using?

I always have an Emacs GUI running, but if I want to do a quick edit on a file I like to use emacsclient in the terminal. However this fails in WezTerm, which is my standard terminal always running:

    $ emacsclient -nw README.md
    emacsclient: could not get terminal name
    $

Emacs then proceeds to open another Emacs GUI.

 I have tried exactly the same on all other terminal emulators I have installed (Alacritty, Kitty, Gnome Terminal, Tilix) and none of them have a problem opening the file in terminal mode.

I CAN open Emacs (not client mode)  terminal mode using the ` -nw`  (or `--no-window`) switch.

- My Emacs version is 29.3
- I am running the Flatpak version of WezTerm.

### WezTerm version

Flatpak 20240203-110809-5046fc22

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

No, and I'll explain why below

### Describe the bug

I am using the flatpak version.

### To Reproduce

Run the flatpak version on a Linux system, I am running Linux Mint Debian Edition 6 "Faye".

### Configuration

No config.

### Expected Behavior

The expected behaviour is to open Emacs in terminal mode.

**Comment by Bihart** [comment]
**tl;dr**
Add `device=all` sandbox permissions to wezterm.

It may be too late, but if you use any non-flatpak version of wezterm, emacsclient will work fine. Why? since wezterm can access `/dev/pty`, one way to test this is to run `tty` command ([the way emaclient works](https://github.com/emacs-mirror/emacs/blob/7dcc05987160cabc0c063bc285d34cdb2c3c600f/lib-src/emacsclient.c#L1124C1-L1125C1)), when running `tty` command in wezterm installed with flatpak (with default permissions) it says `this is not a tty`, how to give wezterm (installed with flatpak) access to `/dev/pty`? Since there is no `device=pty` ([source](https://docs.flatpak.org/en/latest/sandbox-permissions.html)), we need to use `device=all`.
